core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -407,23 +407,13 @@ object H2OContext extends Logging {
 
   private class H2OContextRestAPIBased(spark: SparkSession, conf: H2OConf) extends H2OContext(spark, conf) with H2OContextRestAPIUtils {
     // Once H2O exposes leader node via rest api, remove the nodes argument
-    override protected def getFlowIp(nodes: Array[NodeDesc]): String = nodes.head.hostname
+    override protected def getFlowIp(nodes: Array[NodeDesc]): String = getLeaderNode(nodes).hostname
```

Comment:
note: the list of nodes is at this stage static, however leader node might change during H2O run-time